model/reliability/db_test.go

```diff
@@ -115,15 +69,43 @@ func TestPipeline(t *testing.T) {
 	}
 
 	for opName, opTests := range map[string]func(ctx context.Context, t *testing.T, env evergreen.Environment){
-		"BuildMatchStageForTask": func(ctx context.Context, t *testing.T, env evergreen.Environment) {
+		// Test DateBoundary creation for the task reliability pipeline.
+		"DateBoundaries": func(ctx context.Context, t *testing.T, env evergreen.Environment) {
 			for testName, testCase := range map[string]func(ctx context.Context, t *testing.T, filter *TaskReliabilityFilter){
 				"Simple": func(ctx context.Context, t *testing.T, filter *TaskReliabilityFilter) {
-
-					// Adding 1 day to to before for consistency with TaskReliabilityQueryPipeline.
-					boundaries := []time.Time{before.Add(24 * time.Hour), after}
 					withCancelledContext(ctx, func(ctx context.Context) {
-						stage := filter.BuildMatchStageForTask(boundaries)
-						assert.Equal(t, stage, match)
+						expected := []time.Time{before.Add(24 * time.Hour), before, after}
+						boundaries := filter.DateBoundaries()
+						assert.Equal(t, len(expected), len(boundaries))
+						assert.Equal(t, expected, boundaries)
+					})
+				},
+				"Before": func(ctx context.Context, t *testing.T, filter *TaskReliabilityFilter) {
+					filter.StatsFilter.BeforeDate = before.Add(24 * time.Hour)
+					withCancelledContext(ctx, func(ctx context.Context) {
+						expected := []time.Time{before.Add(2 * 24 * time.Hour), before.Add(24 * time.Hour), before, after}
```

Comment:
I'm confused by DateBoundaries()...why does expected boundary include 4 items here? I thought it's only supposed to include start, end, and duration?

Reply:
Ah, makes sense now. I read too quickly. Thanks!